File: backend/web/models/plaid_item.py
from web import db

class PlaidItem(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    item_id = db.Column(db.String(100), unique=True)
    access_token = db.Column(db.String(100), unique=True)
    client_id = db.Column(db.String, db.ForeignKey('user.client_id'), nullable=False)

    def __init__(self, **kwargs):
        self.item_id = kwargs.get('item_id')
        self.access_token = kwargs.get('access_token')
        self.client_id = kwargs.get('client_id')
    
    def __repr__(self) -> str:
        return f'<PlaidItem {self.item_id}>'

File: src/main/java/com/accountrix/banxi/model/plaid/item/PlaidItem.java
package com.accountrix.banxi.model.plaid.item;

import com.accountrix.banxi.model.user.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name = "plaid_item")
public class PlaidItem {

    @Id
    @Column(name = "item_id", nullable = false, unique = true)
    private String itemID;

    @Column(name = "access_token", nullable = false, unique = true)
    private String accessToken;

    @ManyToOne(optional = false, fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "client_id", nullable = false)
//    @JsonIgnore
    private User linkedUser;

    public PlaidItem() {

    }

    public PlaidItem(String itemID, String accessToken) {
        this.itemID = itemID;
        this.accessToken = accessToken;
    }

    public PlaidItem(String itemID, String accessToken, User linkedUser) {
        this.itemID = itemID;
        this.accessToken = accessToken;
        this.linkedUser = linkedUser;
    }

    public String getItemID() {
        return itemID;
    }

    public void setItemID(String itemID) {
        this.itemID = itemID;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public User getLinkedUser() {
        return linkedUser;
    }

    public void setLinkedUser(User linkedUser) {
        this.linkedUser = linkedUser;
    }
}
